db/db_test.go

```diff
@@ -343,3 +345,29 @@ func TestServerCertificateVerification(t *testing.T) {
 	})
 }
 
+func TestAuthConnection(t *testing.T) {
+	if !testtype.HasTestType(testtype.AWSAuthTestType) && !testtype.HasTestType(testtype.KerberosTestType) {
+		t.SkipNow()
+	}
+	Convey("With an AWS or Keberos auth URI", t, func() {
+		enabled := options.EnabledOptions{URI: true}
+		var uri string;
+		if testtype.HasTestType(testtype.AWSAuthTestType) {
+			uri = os.Getenv("MONGOD");
+		} else {
+			uri = "mongodb://" + kerberosUsername + "@" + kerberosConnection + "/kerberos?authSource=$external&authMechanism=GSSAPI"
+		}
+		fakeArgs := []string{"--uri=" + uri}
+		toolOptions := options.New("test", "", "", "", true, enabled)
+		toolOptions.URI.AddKnownURIParameters(options.KnownURIOptionsReadPreference)
+		_, err := toolOptions.ParseArgs(fakeArgs)
+		if err != nil {
+			panic("Could not parse MONGOD environment variable")
+		}
+		Convey("a connection should succeed", func() {
+			_, err = NewSessionProvider(*toolOptions)
+			So(err, ShouldBeNil)
+		})
```

Comment:
I've moved the connection test from `options_test.go` to `db_test.go` so that I can test the connection with `NewSessionProvider`.